ERCS/erc-7573.md

```diff
@@ -0,0 +1,102 @@
+---
+eip: 7573
+title: Conditional-upon-Transfer-Decryption for DvP
+description: A Proposal for a Lean and Functional Delivery versus Payment
+author: Christian Fries (@cfries), Peter Kohl-Landgraf (@pekola)
+discussions-to: https://ethereum-magicians.org/t/erc-7573-conditional-upon-transfer-decryption-for-delivery-versus-payment/17232
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-12-05
+---
+
+## Abstract
+
+The interfaces model the functional transaction scheme to establish a secure delivery-versus-payment
+across two blockchains, where a) no intermediary is required and b) one of the two chains
+(e.g. the "payment chain") can securely interact with a stateless "decryption oracle".
+
+For convenience, we call the two chains "asset chain" and "payment chain". However, the concept is agnostic to the nature of the tokens managed on the two chains.
+
+The scheme is realized by two smart contracts, one on each chain.
+One smart contract implements the `ILockingContract` interface on one chain (e.g. the "asset chain"), and another smart contract implements the `IDecryptionContract` interface on the other chain (e.g., the "payment chain").
+
+The smart contract on the asset chain, implementing `ILockingContract`, locks the asset until a key is presented that will encrypt to a given value.
+
+The smart contract on the payment chain, implementing `IDecryptionContract`, decrypts one of two keys (via the decryption oracle) conditional to the success or failure of the token transfer initiated
+by the method
+```solidity
+function transferAndDecrypt(bytes32 id, address to, string memory keyEncryptedSuccess, string memory keyEncryptedFailure) external;
+```
+
+The contract  on the asset chain, implementing `ILockingContract`, will lock the token until a key is presented
+releasing the token by calling
+```solidity
+function transferWithKey(bytes32 id, string memory key) external;
+```
+where `key` is the decrypted `keyEncryptedSuccess` or `keyEncryptedFailure`
+
+For the decryption, a stateless decryption oracle is attached to the chain running `IDecryptionContract`.
```

Comment:
Abstracts have to walk a fine line between giving enough technical information to inform the reader, and being succinct enough to fit in a paragraph or two.

This abstract falls on the side of a bit too much detail. Try to shorten this a bit and remove code blocks, while retaining a high-level technical overview.